prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -188,6 +194,72 @@ public async Task<IActionResult> UpdateSiteCompleted(int siteId)
             return NoContent();
         }
 
+        // PUT: api/Sites/5/adjudicator
+        /// <summary>
+        /// Add a site's assigned adjudicator.
+        /// </summary>
+        /// <param name="siteId"></param>
+        /// <param name="adjudicatorId"></param>
+        [HttpPut("{siteId}/adjudicator", Name = nameof(SetSiteAdjudicator))]
+        [Authorize(Policy = AuthConstants.ADMIN_POLICY)]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<Site>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<Site>> SetSiteAdjudicator(int siteId, [FromQuery] int adjudicatorId)
+        {
+            var site = await _siteService.GetSiteAsync(siteId);
+
+            if (site == null)
+            {
+                return NotFound(ApiResponse.Message($"Site not found with id {siteId}."));
+            }
+
+            if ((adjudicatorId != 0 && !await _adminService.AdminExistsAsync(adjudicatorId)))
+            {
+                return NotFound(ApiResponse.Message($"Admin not found with id {adjudicatorId}."));
+            }
+
+            Admin admin = (adjudicatorId != 0)
```

Comment:
Since adjudicatorId is optional, how about a nullable int with HasValue instead?

Reply:
Updated